pkg/plugin/cmd.go

```diff
@@ -787,9 +842,22 @@ func (o *DebugOptions) getAgentPod() *corev1.Pod {
 							Name:      "cgroup",
 							MountPath: "/sys/fs/cgroup",
 						},
+						// containerd client will need to access /var/data, /run/containerd and /run/runc
+						{
+							Name:      "vardata",
+							MountPath: "/var/data",
+						},
+						{
+							Name:      "runcontainerd",
+							MountPath: "/run/containerd",
+						},
+						{
+							Name:      "runrunc",
+							MountPath: "/run/runc",
+						},
 						{
 							Name:             "lxcfs",
-							MountPath:        "/var/lib/lxc/lxcfs",
+							MountPath:        "/var/lib/lxc",
```

Comment:
@markzhang0928 
With /var/lib/lxc/lxcfs, this sequence would fail in the environments I have access to
kubectl apply -f agent_daemonset.yml  <!-- works
kubectl delete debug-agent                  <!-- works
kubectl apply -f agent_daemonset.yml  <!-- fails

The second use of kubectl apply fails because  /var/lib/lxc/lxcfs on the host is in a bad state and containerd is unable to mount it.   A directory listing run on the host shows 
root@kube-host:/# ls -latr /var/lib/lxc
ls: cannot access '/var/lib/lxc/lxcfs': Transport endpoint is not connected
total 8
d?????????  ? ?    ?       ?            ? lxcfs
drwxr-xr-x 45 root root 4096 Apr 25 02:35 ..
drwxr-xr-x  3 root root 4096 Apr 25 14:35 .

By mounting /var/lib/lxc instead of /var/lib/lxc/lxcfs this problem goes away.   And given that I see the proc files under /var/lib/lxc/lxcfs on the host and in the containers it seems to work.

Fwiw, this doc 
https://kubernetes.io/docs/concepts/storage/volumes/
says "This volume mount will receive all subsequent mounts that are mounted to this volume or any of its subdirectories."     So it seems to work because of how subdirectories of a volumes that is either HostToContainer or Bidirectional are handled.